src/gsCore/gsTemplateTools.h

```diff
@@ -134,6 +135,24 @@ struct remove_cv { typedef typename remove_volatile<typename remove_const<T>::ty
 
 template<typename T> struct is_integral: is_integral_base<typename remove_cv<T>::type> {};
 
+template<class T>
+struct make_unsigned;
+
+#define MAKE_UNSIGNED(signed_type)           \
+template<>                                   \
+struct make_unsigned<signed_type> {          \
+    typedef unsigned signed_type type;       \
+};                                           \
+template<>                                   \
+struct make_unsigned(unsigned signed_type) { \
+    typedef unsigned signed_type type;       \
+};
+
+MAKE_UNSIGNED(short)
+MAKE_UNSIGNED(int)
+MAKE_UNSIGNED(long)
+MAKE_UNSIGNED(long long)
+
```

Comment:
better to rename to GISMO_MAKE_UNSIGNED to avoid name clashes.

maybe also good to undefine the macro afterwards

Reply:
ok, will rename.
Btw. a Unittest is also already in unsupported: https://github.com/gismo/unsupported/blob/master/unittests/gsMakeUnsigned.cpp